### ISSUE TITLE
[pickers] `onMonthChange` doesn't call after click to days outside the current month

### ISSUE THREAD

**Issue by Semigradsky** [mui/mui-x#6953]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example: https://codesandbox.io/s/zen-hodgkin-8ijm8m?file=/demo.tsx

Steps:
1. Add `showDaysOutsideCurrentMonth`
2. Click to day outside current month
3. Only `onChange` callback was called. `onMonthChange` wasn't called.

### Current behavior 😯

`onMonthChange` wasn't called.

### Expected behavior 🤔

`onMonthChange` called.

### Context 🔦

By documentation:
> `onMonthChange` - Callback firing on month change

**Comment by flaviendelangle** [comment]
Indeed, `onMonthChange` is only fired when you change the value from the outside of the picker or if you change it from the month view.

That's not very coherent, I'll look on how to improve it :+1:

**Comment by flaviendelangle** [comment]
This PR was not merged yet because we are unclear about what the callback should designate (should it be fired when the month of the value changes or when the month visible on screen changes).

If it's the 2nd one, then sure the PR will cover your use case.

If it's the 1st one, we will probably add a new callback for the visible month on screen.

**Comment by dhulme** [comment]
Ok thank you. Perhaps keeping the current callback for the value, and adding a new `onVisibleMonthChange` or similar could be a solution?

**Comment by flaviendelangle** [comment]
That's one of the approach we can take indeed

**Comment by gentrit-humolli** [comment]
@flaviendelangle this is still not resolved, right?

Though, on date range calendar, it seems to fire the callback on both scenarios.
